NDTensors/src/UnallocatedArrays/src/unallocated_impl.jl

```diff
@@ -0,0 +1,29 @@
+for Typ in (:UnallocatedFill, :UnallocatedZeros)
+  ## Here are functions specifically defined for UnallocatedArrays
+  ## not implemented by FillArrays
+  @eval begin
+    alloctype(A::$Typ) = alloctype(typeof(A))
+    alloctype(::Type{<:$Typ{ElT,N,Axes,Alloc}}) where {ElT,N,Axes,Alloc} = Alloc
+
+    get_index(A::$Typ) = getindex(data(A))
```

Comment:
What is this definition for?

Reply:
I.e. was it meant to be `get_index`? Also why is it only for scalars and not general indices?

Reply:
This was a typo and most likely it should be something like `get_index(A::$Type, i...) = getindex_value(parent(A))`.

Reply:
But what is `get_index`? That's not a function that exists anywhere else, do you mean `getindex`?

Probably in general we should just forward to `parent` with most operations, i.e. `getindex(A::$Typ, I...) = getindex(parent(A), I...)`. That allows us to rely on `FillArrays` as much as possible.

Reply:
I think the best way is code generation with an eval loop, see for example: https://github.com/mtfishman/DataGraphs.jl/blob/cd9d73b22f006a5b8df87023b1d8e036996d708b/src/abstractdatagraph.jl#L86-L91.

And a discussion here: https://discourse.julialang.org/t/forward-all-methods-of-a-structure/11783

Reply:
Ultimately, I don't think there is a better way, since forwarding method should be opt-in, i.e. you should have to list which methods you want to forward since many methods won't make sense or be safe to forward.

However, in practice you shouldn't have to forward many methods, just the minimal interface required for `AbstractFill`/`AbstractZeros`.

A practical guide to that would be the functions I listed here: https://github.com/JuliaArrays/FillArrays.jl/pull/299#issuecomment-1748840968 as well as the definitions of `Zeros`/`Ones` here: https://github.com/JuliaArrays/FillArrays.jl/blob/7036a5005d2a96f79b67ffc9bfd4a1e07517c85e/src/FillArrays.jl#L281-L354.

In principle if you just overload those functions then everything else should "just work" by inheriting the behavior of `AbstractFill`/`AbstractZeros`.

Reply:
It looks like I didn't need the getindex codes and I have much of the functions working so I will reoslve this issue. Please unresolve if there are still things to address. Thanks!